apps/tp_dump.py

```diff
@@ -372,6 +374,19 @@ def main():
             print("Fragment Path:", path)
         data.load_frag(path)
 
+    # Try to find an empty plotting directory
+    plot_iter = 0
+    plot_dir = f"{data.run_id}-{data.file_index}_figures_{plot_iter:04}"
+    while not overwrite and os.path.isdir(plot_dir):
+        plot_iter += 1
+        plot_dir = f"{data.run_id}-{data.file_index}_figures_{plot_iter:04}"
+    print(f"Saving outputs to ./{plot_dir}/")
+    # If overwriting and it does exist, don't need to make it.
+    # So take the inverse to mkdir.
+    if not (overwrite and os.path.isdir(plot_dir)):
+        os.mkdir(plot_dir)
+    os.chdir(plot_dir)
+
```

Comment:
For the future, `tp_dump.py` and `ta_dump.py` should probably be based off some base class to prevent such code-replication. I have no idea when we will have the time to do that though :)

Reply:
Given that these are just dump scripts, I feel fine neglecting this for now. Given the amount of repetition that can be made, a plotting submodule to `trgtools` would likely be useful.